[transport] make the new derooted sock management less chatty

fabbione opened the pull request:
Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione committed:
[transport] make the new derooted sock management less chatty

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione committed:
[transport] fix errno handling

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione committed:
[transport] consistency is not optional :-)

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

chrissie-c reviewed (dismissed):
ACK

wferi commented on libknet/transport_common.c:
```diff
@@ -87,31 +88,31 @@ static int _configure_sockbuf (knet_handle_t knet_h, int sock, int option, int f
 int _configure_common_socket(knet_handle_t knet_h, int sock, uint64_t flags, const char *type)
 {
 	int err = 0, savederrno = 0;
+	int value;
```
Annoyingly enough, this makes the BSD builds fail with unused variable.

fabbione replied:
Yes indeed. the issue is a missing include from transport_common.c :-)

wferi replied:
You certainly know your stuff: found a hidden bug behind a careless one! 👍
If you know that at least one of those conditionals will always be true, then moving the definition to the front won't cause problems. But why is it preferable? A trap for possible other hidden bugs? :)

fabbione replied:
In this specific case, they were both supposed to be true (either in Linux or BSD), but clearly we need to do a better job at noticing those build differences. I think we need to either start printing more stuff at configure time or like the last commit, at runtime.

wferi commented:
Thanks for fixing my bugs!
Wouldn't it make sense to make the logging calls preserve `errno` by themselves? Do we ever need error codes from them?

fabbione commented:
As for errno vs savederrno, it´s not just for the logging calls. It become a coding standard around and I just kept the same style. You can often find in the code:

savederrno = pthread_mutex...

callfoo()
savederrno = errno;
if (err)
  goto out_unlock;

out_unlock:
  pthread_mutex_unlock..
  errno = savederrno; 
  return err;

I simply got used to write it that way for consistency.

fabbione committed:
[transport] add missing include files and really fix 86c81a32cfccde1dea

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione committed:
[transports] add logging around setsockopts that have specific per distro/include requirements

this is helpful to determine if a given option has been built-in and/or activated
properly

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

chrissie-c reviewed (approved):
re-ACK
